[![DOI](https://zenodo.org/badge/427075004.svg)](https://zenodo.org/badge/latestdoi/427075004)

# aDNA-BAMPlotter
## **ANCIENT DNA BAM PLOTTER SCRIPT FOR MICROBIAL GENOMES**

This script produces a coverage plot, an edit distance bar plot and a deamination line plot. The script is designed to display coverage above a specified mapping quality threshold (default 30) as well as without any simultaneously. Regions of the genome with MQ=0 are also highlighted. Input is an indexed not quality filtered BAM file and optionally the misincorporation.txt file generated by mapDamage2 (Jónsson et al. 2013). It calculates edit distances and mapping stats.

**NOT SUITED FOR LARGE GENOMES SUCH AS EUKARYOTE GENOMES!**

# Requirements
You will need the following python packages to run the script:
- python 3
- numpy
- pysam
- pysamstats
- pandas
- matplotlib
- tqdm

# Usage example:
Standard usage:
```
 python aDNA-BAMPlotter.py -b in.bam -d misincorporation.txt -o out.pdf
 ```

Usage while filtering for two specific headers:
```
 python aDNA-BAMPlotter.py -b yp.bam -d misincorporation.txt -i NC_003143.1 NC_003131.1 -o out.pdf
 ```

Usage while changing the quality threshold to 10:
```
 python aDNA-BAMPlotter.py -b in.bam -d misincorporation.txt -q 10 -o out.pdf
 ```

For help use:
```
 python aDNA-BAMPlotter.py -h
 ```

**Example Plot:**
![alt text](https://github.com/MeriamGuellil/aDNA-BAMPlotter/blob/main/example_plot.png?raw=true)

# Cite:
Meriam Guellil. (2021). MeriamGuellil/aDNA-BAMPlotter: aDNA-BAMPlotter (v2.0.0). Zenodo. https://doi.org/10.5281/zenodo.5676093

# References:
Jónsson, Hákon, Aurélien Ginolhac, Mikkel Schubert, Philip L. F. Johnson, and Ludovic Orlando. 2013. “mapDamage2.0: Fast Approximate Bayesian Estimates of Ancient DNA Damage Parameters.” Bioinformatics  29 (13): 1682–84. https://doi.org/10.1093/bioinformatics/btt193.

Walt, Stéfan van der, S. Chris Colbert, and Gaël Varoquaux. 2011. “The NumPy Array: A Structure for Efficient Numerical Computation.” Computing in Science & Engineering 13 (2): 22–30. https://doi.org/10.1109/MCSE.2011.37.

Pysam: Pysam Is a Python Module for Reading and Manipulating SAM/BAM/VCF/BCF Files. It’s a Lightweight Wrapper of the Htslib C-API, the Same One That Powers Samtools, Bcftools, and Tabix. n.d. Github. https://github.com/pysam-developers/pysam.

Miles, Alistair. n.d. Pysamstats. Github. https://github.com/alimanfoo/pysamstats.

McKinney, Wes, and Others. 2010. “Data Structures for Statistical Computing in Python.” In Proceedings of the 9th Python in Science Conference, 445:51–56. Austin, TX. http://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf.

Hunter, J. D. 2007. “Matplotlib: A 2D Graphics Environment.” Computing in Science Engineering 9 (3): 90–95. https://doi.org/10.1109/MCSE.2007.55.

Tqdm: A Fast, Extensible Progress Bar for Python and CLI. n.d. Github. https://github.com/tqdm/tqdm.



